platform/mds/mds/src/main/java/org/motechproject/mds/ex/object/ObjectCreateException.java

```diff
@@ -22,7 +24,13 @@ private static String getMessageFromCause(Throwable cause) {
         if (cause instanceof ConstraintViolationException) {
             Set<ConstraintViolation<?>> violations = ((ConstraintViolationException) cause).getConstraintViolations();
             for (ConstraintViolation violation : violations) {
-                message += String.format("Field %s %s\n", violation.getPropertyPath(), violation.getMessage() );
+                message += String.format("Field %s %s\n", violation.getPropertyPath(), violation.getMessage());
+            }
+        } else if (cause instanceof JDODataStoreException) {
+            for (Throwable exception : ((JDODataStoreException) cause).getNestedExceptions()) {
+                if (exception instanceof SQLIntegrityConstraintViolationException) {
+                    message += exception.getMessage();
```

Comment:
We don't want to be concatenating raw exception messages (using += no less), we want to display a localizable message for the user.